archunit/src/main/java/com/tngtech/archunit/core/domain/JavaMember.java

```diff
@@ -48,15 +48,14 @@
 
     private final String name;
     private final String descriptor;
-    private final Supplier<Map<String, JavaAnnotation<JavaMember>>> annotations;
+    private Map<String, JavaAnnotation<JavaMember>> annotations = Collections.emptyMap();
```

Comment:
Because we now reset this field in `completeAnnotations(..)`, unfortunately that is the price to pay :man_shrugging: We can have either a final mutable collection or a non-final immutable one and I went for the latter :wink: